// @freakycoder/react-native-helpers/build/dist/utils/index.js
export const capitalizeFirstLetter = (str) => str && str.length ? str.charAt(0).toUpperCase() + str.slice(1) : str;
export const capitalizeAllStartingWords = (str, lower = false) => (lower ? str.toLowerCase() : str).replace(/(?:^|\s|[''({])+\S/g, (match) => match.toUpperCase());
export const capitalize = (str) => str.charAt(0).toUpperCase() + str.slice(1);
export const generateRandomNumber = (min, max) => Math.floor(Math.random() * (max - min + 1) + min);
export const diffArrays = (arr1, arr2) => arr1 &&
    arr2 &&
    arr1.filter((obj1) => !arr2.find((obj2) => obj1.id === obj2.id && obj2.isChecked));
export const diffArraysWithId = (arr1, arr2) => arr1 &&
    arr2 &&
    arr1.filter((obj1) => !arr2.find((obj2) => obj1.id === obj2.id && obj2.id));
export const diffDates = (date, date2) => Math.floor(Math.abs(date - date2) / 86400000);
export const isBlankString = (str) => !str || str.length === 0 || /^\s*$/.test(str);
export const randomBoolean = () => Math.random() >= 0.5;
export const coinflip = () => Math.random() >= 0.5;
export const isEven = (val) => val % 2 === 0;
export const removeAllDuplicateValuesInArray = (arr) => [
    ...new Set(arr),
];
export const isArray = (arr) => Array.isArray(arr);
export const generateRandomString = () => Math.random().toString(36).slice(2);
export const mergeArrays = (a, b) => [...a, ...b];
export const mergeArraysAndRemoveDuplicates = (a, b) => [
    ...new Set([...a, ...b]),
];
export const getTrueType = (obj) => Object.prototype.toString.call(obj).slice(8, -1).toLocaleLowerCase();
export const isNotEmptyArray = (arr) => Array.isArray(arr) && arr.length > 0;
export const safelyParseJson = (string) => {
    try {
        return JSON.parse(string);
    }
    catch {
        return string;
    }
};
export const isEmptyObj = (obj) => Reflect.ownKeys(obj).length === 0 && obj.constructor === Object;
export const shuffleArray = (arr) => arr.sort(() => 0.5 - Math.random());
export const convertSnakeToCamelCase = (str) => {
    return str.replace(/([-_][a-z])/g, (group) => group.toUpperCase().replace("-", "").replace("_", ""));
};
export const getRandomHexColor = () => `#${Math.floor(Math.random() * 0xffffff)
    .toString(16)
    .padEnd(6, "0")}`;
export const convertRGBToHexColor = (r, g, b) => "#" + ((1 << 24) + (r << 16) + (g << 8) + b).toString(16).slice(1);
export const getMinMaxOfArray = (arr) => [
    Math.min(...arr),
    Math.max(...arr),
];
export const sleep = (delay) => {
    return new Promise((resolve) => setTimeout(resolve, delay));
};
/**
 * Create an object from the pairs of key and value
 * @param arr
 * @returns
 */
export const toObj = (arr) => Object.fromEntries(arr);
/**
 * Get union of arrays
 * @param arr
 * @returns
 */
export const getUnion = (...arr) => [...new Set(arr.flat())];
/**
 * Partition an array based on the criteria/condition
 * @param arr
 * @param criteria
 * @returns
 */
export const partition = (arr, criteria) => arr.reduce((acc, i) => (acc[criteria(i) ? 0 : 1].push(i), acc), [[], []]);
